consensus/misc/eip4844/eip4844.go

```diff
@@ -57,27 +71,86 @@ func VerifyEIP4844Header(parent, header *types.Header) error {
 		parentExcessBlobGas = *parent.ExcessBlobGas
 		parentBlobGasUsed = *parent.BlobGasUsed
 	}
-	expectedExcessBlobGas := CalcExcessBlobGas(parentExcessBlobGas, parentBlobGasUsed)
-	if *header.ExcessBlobGas != expectedExcessBlobGas {
-		return fmt.Errorf("invalid excessBlobGas: have %d, want %d, parent excessBlobGas %d, parent blobDataUsed %d",
-			*header.ExcessBlobGas, expectedExcessBlobGas, parentExcessBlobGas, parentBlobGasUsed)
+	excessBlobGas := parentExcessBlobGas + parentBlobGasUsed
+	targetGas := uint64(targetBlobsPerBlock(config, headTimestamp)) * params.BlobTxBlobGasPerBlob
+	if excessBlobGas < targetGas {
+		return 0
 	}
-	return nil
+	return excessBlobGas - targetGas
 }
 
-// CalcExcessBlobGas calculates the excess blob gas after applying the set of
-// blobs on top of the excess blob gas.
-func CalcExcessBlobGas(parentExcessBlobGas uint64, parentBlobGasUsed uint64) uint64 {
-	excessBlobGas := parentExcessBlobGas + parentBlobGasUsed
-	if excessBlobGas < params.BlobTxTargetBlobGasPerBlock {
+// CalcBlobFee calculates the blobfee from the header's excess blob gas field.
+func CalcBlobFee(config *params.ChainConfig, header *types.Header) *big.Int {
+	var frac uint64
+	switch config.LatestFork(header.Time) {
```

Comment:
We should add an is-optimism check here, and return `minBlobGasPrice` if yes.
You can verify it with:
```
func TestValue(t *testing.T) {
	excessBlobGas := big.NewInt(0)
	frac := new(big.Int).SetUint64(params.DefaultCancunBlobConfig.UpdateFraction)
	v := fakeExponential(minBlobGasPrice, excessBlobGas, frac)
	t.Logf("value: %d", v)
}
```
It's always 1. Not really used, since there are no blobs included.

Reply:
Just need something like this, then rerun monorepo PR CI, and then if green, ready to merge I think.

Reply:
Thanks, came up with the same fix. I also check additionally that the blob schedule is nil, and that the excess blob field is set to 0, so we apply the short circuit only to blocks that have the expected excess blob field set. Although I'm wondering about the Cancun transition block - should I change the check to
```
	if config.IsOptimism() && config.BlobScheduleConfig == nil
		&& (header.ExcessBlobGas == nil || *header.ExcessBlobGas == 0) {
		return minBlobGasPrice
	}
```
?

Reply:
Nvm I see that `CalcBlobFee` is only called if the `header.ExcessBlobGas` is guaranteed to be non-nil.